<?xml version="1.0"?>
<launch>
    <!-- Square mode launch file for encoder_pos_node -->
    
    <!-- Vehicle name parameter -->
    <arg name="veh" default="$(env VEHICLE_NAME)" doc="Name of the vehicle"/>
    
    <!-- Encoder Pose Node in Square Mode -->
    <node name="encoder_pos_node_square" 
          pkg="my_package" 
          type="encoder_pos_node.py" 
          output="screen"
          clear_params="true">
        
        <!-- Enable square mode -->
        <param name="square_mode" value="true"/>
        
        <!-- Remap topics to vehicle-specific names -->
        <remap from="~car_cmd" to="joy_mapper_node/car_cmd"/>
        <remap from="~shutdown" to="shutdown"/>
        
        <!-- Parameters -->
        <param name="veh" value="$(arg veh)"/>
        
    </node>
    
</launch>
